src/sdk/gateway/gateway.service.ts

```diff
@@ -608,6 +610,92 @@ export class GatewayService extends Service {
     return result;
   }
 
+  async cancelGatewayBatch(hash: string): Promise<GatewaySubmittedBatch> {
+    const { accountService, apiService } = this.services;
+
+    const { accountAddress } = accountService;
+
+    const { result } = await apiService.mutate<{
+      result: GatewaySubmittedBatch;
+    }>(
+      gql`
+        mutation($chainId: Int, $account: String!, $hash: String!) {
+          result: cancelGatewayBatch(chainId: $chainId, account: $account, hash: $hash) {
+            hash
+            state
+            account
+            nonce
+            to
+            data
+            senderSignature
+            estimatedGas
+            estimatedGasPrice
+            feeToken
+            feeAmount
+            feeData
+            delayedUntil
+            createdAt
+            updatedAt
+          }
+        }
+      `,
+      {
+        models: {
+          result: GatewaySubmittedBatch,
+        },
+        variables: {
+          account: accountAddress,
+          hash,
+        },
+      },
+    );
+
+    return result;
+  }
+
+  async forceGatewayBatch(hash: string): Promise<GatewaySubmittedBatch> {
+    const { accountService, apiService } = this.services;
+
+    const { accountAddress } = accountService;
+
+    const { result } = await apiService.mutate<{
+      result: GatewaySubmittedBatch;
+    }>(
+      gql`
+        mutation($chainId: Int, $account: String!, $hash: String!) {
+          result: forceGatewayBatch(chainId: $chainId, account: $account, hash: $hash) {
+            hash
```

Comment:
is logs field excluded intentionally here and above?

Reply:
Intentionally, since there are no logs for pending batches